src/middlewared/middlewared/plugins/enclosure_/jbof_enclosures.py

```diff
@@ -0,0 +1,159 @@
+        # value they would have if a device was
+        # inserted into a proper SES device (or not).
+        # Since this is NVMe (which deals with PCIe)
+        # that paradigm doesn't exist per se but we're
+        # "faking" a SES device, hence the hex values.
+        # The `status` variables use same logic.
+        if device is not None:
+            status = 'OK'
+            value_raw = 0x1000000
+        else:
+            status = 'Not installed'
+            value_raw = 0x5000000
+
+        mapped_slot = disks_map['versions']['DEFAULT']['model'][model][slot]['mapped_slot']
+        fake_enclosure['elements']['Array Device Slot'][mapped_slot] = {
+            'descriptor': f'Disk #{slot}',
+            'status': status,
+            'value': None,
+            'value_raw': value_raw,
+            'dev': device,
+            'original': {
+                'enclosure_id': uuid,
+                'enclosure_sg': None,
+                'enclosure_bsg': None,
+                'descriptor': f'slot{slot}',
+                'slot': slot,
+            }
+        }
+
+    return [fake_enclosure]
+
+
+def map_es24n(model, rclient, uri):
+    try:
+        all_disks = rclient.get(f'{uri}/Drives?$expand=*').json()
+    except Exception:
+        LOGGER.error('Unexpected failure enumerating all disk info', exc_info=True)
+        return
+
+    num_of_slots = all_disks['Members@odata.count']
+    ui_info = {
+        'rackmount': True,
+        'top_loaded': False,
+        'front_slots': num_of_slots,
+        'rear_slots': 0,
+        'internal_slots': 0
+    }
+    mounted_disks = {
+        v['serial']: (k, v) for k, v in get_sys_class_nvme().items()
+        if v['serial'] and v['transport_protocol'] == 'rdma'
+    }
+    mapped = dict()
+    for disk in all_disks['Members']:
+        slot = int(disk['Id'])
+        if found := mounted_disks.get(disk['SerialNumber']):
+            mapped[slot] = found[0]
+        else:
+            mapped[slot] = None
+
+    return fake_jbof_enclosure(model, rclient.uuid, num_of_slots, mapped, ui_info)
+
+
+def get_redfish_clients(jbofs):
+    clients = dict()
+    for jbof in jbofs:
+        try:
+            rclient = RedfishClient(
+                f'https://{jbof["mgmt_ip1"]}', jbof['mgmt_username'], jbof['mgmt_password']
+            )
+            clients[jbof['mgmt_ip1']] = rclient
+        except InvalidCredentialsError:
+            LOGGER.error('Failed to login to redfish ip %r', jbof['mgmt_ip1'])
+        except Exception:
+            LOGGER.error('Unexpected failure creating redfish client object', exc_info=True)
+
+    return clients
+
+
+def get_enclosure_model(rclient):
+    model = uri = None
+    try:
+        chassis = rclient.chassis()
+    except Exception:
+        LOGGER.error('Unexpected failure enumerating chassis info', exc_info=True)
+        return model, uri
+
+    try:
+        for _, uri in chassis.items():
+            info = rclient.get(uri)
```

Comment:
I take it you're iterating over things and grabbing the '`Model`" because you don't trust the name '`2U24`' & want to make the code more generally applicable - a noble goal.

One issue with this is that with the ES24n the 3 children of Chassis are `2U24`, `IOM1` & `IOM2` and they **ALL** have a `Model` component, so the current code is succeeding by luck of ordering - so not really generally applicable.
```
>>> rclient.get('/redfish/v1/Chassis/2U24').json()['Model']
'VDS2249R2'
>>> rclient.get('/redfish/v1/Chassis/IOM1').json()['Model']
'X'
>>> rclient.get('/redfish/v1/Chassis/IOM2').json()['Model']
'X'
```

I wonder whether using the Product from the [ServiceRoot schema](https://redfish.dmtf.org/schemas/v1/ServiceRoot.v1_16_1.json) might be easier & more robust.
```
>>> rclient.get('/redfish/v1').json()['Product']
'ves-vds2249r-RFlex'
```

At least it **is** in the documented Redfish schema.

Reply:
Bummer, great catch. I'll check up on this

Reply:
Wait, no this is wrong. I account for this in the next few lines. The `IOM` entries also have a model but their model value isn't in our `JbofModels` enum.
```
            if info.ok:
                 try:
                     model = JbofModels(info.json().get('Model', '')).name
                     break
                 except KeyError:
                     continue